math/min/min_test.go

```diff
@@ -1,81 +1,62 @@
-package min_test
+package min
 
 import (
 	"testing"
-
-	"github.com/TheAlgorithms/Go/math/min"
 )
 
-func TestMin(t *testing.T) {
-	testCases := []struct {
-		name  string
-		left  int
-		right int
-		min   int
+func getTestCases() []struct {
```

Comment:
> I do not think this is how @tjgurwara99 wanted you to implement it. See [this](https://github.com/TheAlgorithms/Go/blob/master/sort/sorts_test.go) for example.

IMHO, what I did covers a specific task and is generally similar to this example

Reply:
The thing is, you will have to rewrite the range loop that  you are using to go through the test cases every time you add a new implementation. In my provided example, you just pass your new implementation to a function, and thats it.

Reply:
> The thing is, you will have to rewrite the range loop that you are using to go through the test cases every time you add a new implementation. In my provided example, you just pass your new implementation to a function, and thats it.

yes, I understand the idea, and generally agree with you, but it seems that there will not be so many algorithms for finding the minimum number, and for this case it is an overhead

Reply:
But if the majority votes that this is very important and necessary and will make us all happier, I will correct it.

Reply:
@tjgurwara99 @siriak You opinions on this matter?

Reply:
I this situation, a framework would not work because the specific bitwise min algorithm requires a base so the function signature is different. I agree in general we should have a framework function that takes the function as an argument but it would not work in this particular case so IMO its fine for now. What I asked for was a test case generator function so that the tests can work on the same cases instead of writing new one's every time and I think this PR covers it now. If a new min/max algorithm gets submitted and has the same signature as vanilla min/max then it would make sense to add a framework.

Reply:
Well in that case, there is no need to use a function for the test cases, a global variable should work fine.

Reply:
> Well in that case, there is no need to use a function for the test cases, a global variable should work fine.

What's the difference? in my opinion, in this case it doesn't matter how we get test cases, it doesn't affect what. What is the real profit?

Reply:
plus, if in the future new implementations of the algorithm appear, then the function will be more flexible

Reply:
> Well in that case, there is no need to use a function for the test cases, a global variable should work fine.

True 😅 I thought it would be cleaner to have a function do the setup and not have any global variables.